scripts/proto/lbann_proto.py

```diff
@@ -371,34 +372,37 @@ def proto(self, model):
     _create_proto_callback(field.message_type.name, field.name)
 _add_to_module_namespace(_proto_callbacks)
 
-def traverse_model(first_module, func):
-    """Traverse a model and apply func to each layer.
+def traverse_layer_graph(start_layers):
```

Comment:
Going this route, I think it would be better to make this a generator instead of explicitly returning a list. It's more Pythonic.

Reply:
Done. Our traversals may be a bit heavy-weight (DFS to find root nodes, DFS for topologically sorted traversal), but I think safety is more important than performance here.